### DISCUSSION TITLE
python 在土木工程上的應用

### DISCUSSION THREAD

**Discussion by YANRONG0123** [1101-GS4719-C/Python-Community-GS4719-C#1940] [科技新知]
隨著時代的演進，python也能夠應用在傳統產業上了，相信蠻多人對於土木系，或是相關的產業，都是有著不好印象，造成不僅僅是工人短缺，還導致土木相關工程師也短缺。

　　但是我們可以利用python，進行更加準確的運算，例如使用文章中提到的Numpy進行力學設計上的分析，或是利用Pandas使現地探勘到的資料視覺化，也能用requests程式庫將繪圖傳送到虛擬的API，實現與專案管理工具或協作平台的整合。

　　藉由python，可以使資料達到自動化的處理，也能降低計算上的錯誤，更可以減少人事成本，解決目前土木工程界的問題。

文章連結: https://mysteryweevil.medium.com/python-for-civil-engineers-streamlining-calculations-and-automating-workflows-47aa607c6728

**Comment by LinChloe** [comment]
我認識的許多土木系朋友都也不知道大一上為什麼需要學程式語言，如果給他們看了你的文章相信會更努力學習程式。
話說我們學校的土木系是教C++，而不是python。
這表示他們需要來上這門通識課(如果搶得到的話~)

**Reply by stone971275** [reply]
其實系上有開一堂數值分析，想更深入的知道如何應用在土木工程上，可以去修看看，老師也沒限制語法。

**Comment by thomasxshen** [comment]
我本身也是土木系的，跟你一樣，我覺得土木工程系的進步日新月異，python已經是所有工程學群必備的程式語言，我在這邊推薦學弟妹一定要把python學好，不要等到了研究所還在學python，超級可悲！像是系上林遠見教授所開的python課程就很值得各位去學習，可以學習如何進行資料前處理。
